wire-compiler/src/main/java/com/squareup/wire/MessageWriter.java

```diff
@@ -147,7 +147,7 @@ public void emitType(JavaWriter writer, TypeElement type, String currentType,
         parameters.add(trailingSegment(option.name));
       }
 
-      writer.beginConstructor(EnumSet.of(PRIVATE), parameters, null);
+      writer.beginConstructor(Collections.<Modifier>emptySet(), parameters, null);
```

Comment:
I had that originally but then was curious of its behavior. It allocates a mutable set with the capacity to hold all of the enum values and makes a copy of the enum constant array. Granted, this is the JVM so it's hard to nitpick so much, but I'm a trained allocation killer from doing Android.